test/e2e/elasticache/tests/test_replicationgroup_largecluster.py

```diff
@@ -0,0 +1,75 @@
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Large cluster load test for replication group resource
+"""
+
+import pytest
+
+from time import sleep
+from common import k8s
+from elasticache.tests.test_replicationgroup import make_replication_group, rg_deletion_waiter, make_rg_name, DEFAULT_WAIT_SECS
+from elasticache.util import provide_node_group_configuration
+
+DEFAULT_WAIT_SECS = 30
+
+@pytest.fixture(scope="module")
+def rg_largecluster_input(make_rg_name):
+    return {
+        "RG_ID": make_rg_name("rg-load-test"),
+        "NUM_NODE_GROUPS": "50",
+        "REPLICAS_PER_NODE_GROUP": "1"
+    }
+
+@pytest.fixture(scope="module")
+def rg_largecluster(rg_largecluster_input, make_replication_group, rg_deletion_waiter):
+    input_dict = rg_largecluster_input
+
+    (reference, resource) = make_replication_group("replicationgroup_loadtest", input_dict, input_dict["RG_ID"])
+    yield (reference, resource)
+
+    # teardown
+    k8s.delete_custom_resource(reference)
+    sleep(DEFAULT_WAIT_SECS)
+    rg_deletion_waiter.wait(ReplicationGroupId=input_dict["RG_ID"])
+
+class TestReplicationGroupLargeCluster:
+
+    @pytest.mark.slow
+    def test_rg_largecluster(self, rg_largecluster_input, rg_largecluster):
+        (reference, _) = rg_largecluster
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True", wait_periods=60)
+
+        # assertions after initial creation
+        desired_node_groups = int(rg_largecluster_input['NUM_NODE_GROUPS'])
+        desired_replica_count = int(rg_largecluster_input['REPLICAS_PER_NODE_GROUP'])
+        desired_total_nodes = (desired_node_groups * (1 + desired_replica_count))
+        resource = k8s.get_resource(reference)
+        assert resource['status']['status'] == "available"
+        assert len(resource['status']['nodeGroups']) == desired_node_groups
+        assert len(resource['status']['memberClusters']) == desired_total_nodes
+
+        # update, wait for resource to sync
+        desired_node_groups = desired_node_groups - 10
+        desired_total_nodes = (desired_node_groups * (1 + desired_replica_count))
+        patch = {"spec": {"numNodeGroups": desired_node_groups,
+                          "nodeGroupConfiguration": provide_node_group_configuration(desired_node_groups)}}
+        _ = k8s.patch_custom_resource(reference, patch)
+        sleep(DEFAULT_WAIT_SECS) # required as controller has likely not placed the resource in modifying
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True", wait_periods=90)
+
+        # assert new state after scaling in
+        resource = k8s.get_resource(reference)
+        assert resource['status']['status'] == "available"
+        assert len(resource['status']['nodeGroups']) == desired_node_groups
+        assert len(resource['status']['memberClusters']) == desired_total_nodes
```

Comment:
I thought from the name of this test that it would be testing the load on the ACK controller, but that isn't actually the case here; instead this is simply testing that the ACK controller waits for a long-ish period of time for a large ElastiCache cluster to get into a available state. This is fine, but it's not a load test in any sense of the word. Perhaps consider changing any language in the PR commit message and files like `test/e2e/elasticache/resources/replicationgroup_loadtest.yaml` to remove the word "load".

Reply:
I suppose that's true, could you clarify what a load test might look like? Something like making a large volume of requests to the controller and measuring some metric?

I think this test was just to check that the controller doesn't have any issues managing the largest cluster an Elasticache customer can create (with respect to pagination or any other issues that might arise). The long wait times were more so a side effect of creating and modifying a large cluster rather than the main purpose of the test, but I agree that it doesn't necessarily "load" the controller much more than any other regular test.